### PR TITLE
[INLONG-6751][Sort] Add read phase metric and table level metric for Oracle

### LINKED ISSUE #6751: [Improve][Sort] Add read phase metric and table level metric for Oracle
### Description

Add read phase metric and table level metric for Oracle. 

### InLong Component

InLong Sort

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by e-mhui**
Add read phase metric and table level metric for Oracle

- Fixes #6751 

### Motivation

Add read phase metric and table level metric for Oracle

### Modifications

*Describe the modifications you've done.*

### CODE REVIEW

**Comment by gong** [comment]
@e-mhui oracle need database-schema-table level metric, pr just add database-table. @chestnut-c please help review

**Comment by e-mhui** [comment]
> @e-mhui oracle need database-schema-table level metric, pr just add database-table. @chestnut-c please help review

ok，I‘ll add it.

**Review by healchow — APPROVED** [review]

**Review by gong — APPROVED** [review]

File: inlong-sort/sort-connectors/base/src/main/java/org/apache/inlong/sort/base/metric/sub/SourceTableMetricData.java [hunk 1]
```diff
@@ -122,9 +122,14 @@ private SourceMetricData buildSubSourceMetricData(String[] schemaInfoArray, Metr
...
-                .append(DELIMITER).append(Constants.TABLE_NAME).append("=").append(schemaInfoArray[1]);
-
+        if (schemaInfoArray.length == 2) {
+            labelBuilder.append(DELIMITER).append(Constants.DATABASE_NAME).append("=").append(schemaInfoArray[0])
+                    .append(DELIMITER).append(Constants.TABLE_NAME).append("=").append(schemaInfoArray[1]);
+        } else if (schemaInfoArray.length == 3) {
+            labelBuilder.append(DELIMITER).append(Constants.DATABASE_NAME).append("=").append(schemaInfoArray[0])
+                    .append(DELIMITER).append(Constants.SCHEMA_NAME).append("=").append(schemaInfoArray[1])
+                    .append(DELIMITER).append(Constants.TABLE_NAME).append("=").append(schemaInfoArray[2]);
+        }
```
Comment (yunqingmoswu) [inline inlong-sort/sort-connectors/base/src/main/java/org/apache/inlong/sort/base/metric/sub/SourceTableMetricData.java:132]: Maybe add a validation is better?

**Review by EMsnap — APPROVED** [review]

File: inlong-sort/sort-connectors/base/src/main/java/org/apache/inlong/sort/base/metric/sub/SourceTableMetricData.java (hunk 1 above)
Comment (e-mhui) [reply inlong-sort/sort-connectors/base/src/main/java/org/apache/inlong/sort/base/metric/sub/SourceTableMetricData.java:132]: There is no need to validate here. The `schemaInfoArray` has been validated before it's initialization. The code is as follows:

```java
    public void outputMetricsWithEstimate(String database, String schema, String table,
            boolean isSnapshotRecord, Object data) {
        if (StringUtils.isBlank(database) || StringUtils.isBlank(schema) || StringUtils.isBlank(table)) {
            outputMetricsWithEstimate(data);
            return;
        }
        String identify = buildSchemaIdentify(database, schema, table);
        SourceMetricData subSourceMetricData;
        if (subSourceMetricMap.containsKey(identify)) {
            subSourceMetricData = subSourceMetricMap.get(identify);
        } else {
            subSourceMetricData = buildSubSourceMetricData(new String[]{database, schema, table}, this);
            subSourceMetricMap.put(identify, subSourceMetricData);
        }
        // source metric and sub source metric output metrics
        long rowCountSize = 1L;
        long rowDataSize = data.toString().getBytes(StandardCharsets.UTF_8).length;
        this.outputMetrics(rowCountSize, rowDataSize);
        subSourceMetricData.outputMetrics(rowCountSize, rowDataSize);

        // output read phase metric
        outputReadPhaseMetrics((isSnapshotRecord) ? ReadPhase.SNAPSHOT_PHASE : ReadPhase.INCREASE_PHASE);
    }
```

**Pull request merged**